frame/staking-rewards/src/lib.rs

```diff
@@ -236,29 +256,48 @@ pub mod pallet {
 			+ Into<u128>
 			+ Zero;
 
-		/// The position id type.
-		type PositionId: Parameter + Member + Clone + FullCodec + Copy + Zero + One + SafeArithmetic;
-
 		type AssetId: Parameter
 			+ Member
 			+ AssetIdLike
 			+ MaybeSerializeDeserialize
 			+ Ord
 			+ From<u128>
-			+ Into<u128>;
-
-		type FinancialNft: nonfungibles::Mutate<AccountIdOf<Self>>
-			+ Create<AccountIdOf<Self>, CollectionId = Self::AssetId>
-			+ FinancialNft<AccountIdOf<Self>, CollectionId = Self::AssetId>;
+			+ Into<u128>
+			+ Copy;
+
+		// REVIEW(benluelo): Mutate::CollectionId type?
+		type FinancialNft: NonFungiblesMutate<AccountIdOf<Self>>
+			+ NonFungiblesCreate<
+				AccountIdOf<Self>,
+				CollectionId = Self::AssetId,
+				ItemId = Self::FinancialNftInstanceId,
+			> + FinancialNft<
+				AccountIdOf<Self>,
+				CollectionId = Self::AssetId,
+				ItemId = Self::FinancialNftInstanceId,
+			>;
+
+		// https://github.com/rust-lang/rust/issues/52662
```

Comment:
:confused:

Reply:
yeah, totally forgot about this limitation. hopefully it's fixed soon :sweat_smile